### PR TITLE
Added a time-averaging plugin.

### PR DESCRIPTION
**Pull request by pv101**
Note that 'stats' are different to those added by the soon to be depreciated time-averaging utility. We can discuss what is most appropriate going forwards (in the context of the proposed improvements to this time-averaging plugin).

I have tested with the 2D Euler Vortex. Should probably test further before accepting.

### CODE REVIEW

File: pyfr/plugins/tavg.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                a += 0.5*(intg.tcurr - self.pt)*(p + c)
+
+            self.psoln = [s.copy() for s in intg.soln]
+            self.pt = intg.tcurr
+
+            if dowrite:
+                self.asoln = [a / self.dtout for a in self.asoln]
+
+                stats = Inifile()
+                stats.set('tavg', 'start', intg.tcurr - self.dtout)
```
Comment (FreddieWitherden) [inline pyfr/plugins/tavg.py]: Do we want to include the regular stats here, too?

**Comment by FreddieWitherden** [comment]
It is a bit light on comments/documentation although given I will be extending it as soon as we merge that is probably fine.

File: pyfr/plugins/tavg.py (hunk 1 above)
Comment (pv101) [inline pyfr/plugins/tavg.py]: I don't think they (tcurr, nsteps, nfevals) make much sense for an averaged solution, so I elected to omit for the moment. We can discuss exactly what to include going forwards when your extension to this is merged?

**Comment by pv101** [comment]
Yes, I think comment situation is fine given it will be extended soon after merging.

**Pull request merged**